ch05/03_bonus_pretraining_on_gutenberg/prepare_dataset.py

```diff
@@ -66,7 +77,7 @@ def combine_files(file_paths, target_dir, max_size_mb=500, separator="<|endoftex
     args = parser.parse_args()
 
     all_files = [os.path.join(path, name) for path, subdirs, files in os.walk(args.data_dir)
-                 for name in files if name.endswith((".txt", ".txt.utf8")) and "raw" not in path]
+                 for name in files if name.endswith((".txt", ".txt.utf8"))]
```

Comment:
Just a small question here, why did you remove the `"raw" not in path` check? Is this because you applied the code in two stages, I.e., the regular preparation and then this on the prepared data?

Reply:
Thanks for pointing that out—it was my mistake.

I was concerned that someone might carelessly execute [gutenberg/process_data.py](https://github.com/pgcorpus/gutenberg/blob/master/process_data.py) like I did, leading to issues like  https://github.com/rasbt/LLMs-from-scratch/discussions/293, so I set the default value to gutenberg/data/.mirror. 

but I forgot that the data_dir modified to gutenberg/data/.mirror would be overridden by the command in [tutorial(Prepare the dataset)](https://github.com/rasbt/LLMs-from-scratch/tree/main/ch05/03_bonus_pretraining_on_gutenberg#2-prepare-the-dataset) under shell:
```shell
python prepare_dataset.py \
  --data_dir gutenberg/data \
  --max_size_mb 500 \
  --output_dir gutenberg_preprocessed
```
`"raw" not in path` still need if the shell argument is `gutenberg/data`.

Reply:
Oh, I think I understand this now. Thanks!

I agree with changing `--data_dir gutenberg/data` to `--data_dir gutenberg/data/raw \`, it makes sense to me!

PS: I also added a progress bar to get a better idea of how long it takes.